persistence/javadsl/src/test/scala/com/lightbend/lagom/javadsl/persistence/AbstractReadSideSpec.scala

```diff
@@ -37,46 +48,75 @@ trait AbstractReadSideSpec extends ImplicitSender { spec: ActorSystemSpec =>
 
   def getAppendCount(id: String): CompletionStage[java.lang.Long]
 
-  private def assertSelectCount(id: String, expected: Long): Unit = {
-    within(20.seconds) {
-      awaitAssert {
-        val count = Await.result(getAppendCount(id).toScala, 5.seconds)
-        count should ===(expected)
-      }
-    }
+  private val tag = TestEntity.Evt.AGGREGATE_EVENT_SHARDS.forEntityId("1")
+
+  private def createTestEntityRef() = {
+    system.actorOf(
+      PersistentEntityActor.props(
+        "test",
+        Optional.of("1"),
+        () => new TestEntity(system),
+        Optional.empty(),
+        10.seconds
+      )
+    )
   }
 
-  private def createReadSideProcessor(tag: AggregateEventTag[TestEntity.Evt]) = {
+  private def createReadSideProcessor() = {
     /* read side and injector only needed for deprecated register method */
-    val readSide = system.actorOf(ReadSideActor.props[TestEntity.Evt](
-      processorFactory,
-      eventStream, classOf[TestEntity.Evt], new ClusterStartupTask(testActor), 20.seconds
-    ))
+    val readSide = system.actorOf(
+      ReadSideActor.props[TestEntity.Evt](
+        processorFactory,
+        eventStream,
+        classOf[TestEntity.Evt],
+        new ClusterStartupTask(testActor),
+        20.seconds
+      )
+    )
 
     readSide ! EnsureActive(tag.tag)
 
     expectMsg(Execute)
 
-    processorFactory().buildHandler().globalPrepare().toScala.foreach { _ =>
-      readSide ! Done
-    }
+    processorFactory()
+      .buildHandler()
+      .globalPrepare().toScala
+      .foreach { _ =>
+        readSide ! Done
+      }
 
     readSide
   }
 
+  private def assertSelectCount(id: String, expected: Long): Unit = {
+    eventually {
+      val count = getAppendCount(id).toScala.futureValue
+      count shouldBe expected
+    }
+  }
+
+  private def fetchLastOffset(): O =
+    processorFactory()
+      .buildHandler()
+      .prepare(tag)
+      .toScala
+      .map(_.asInstanceOf[O]) // no ClassTag, no mapTo :-(
```

Comment:
I guess it comes from my initial wrong assumption that is was always a `Long`. I had a `mapTo[Long]` that I had to removed because cassandra was returning a `TimeBasedUUID`. 

But you are right, we just keep it as `Any`. That's all private and internal usage.